tinydraw_test.go

```diff
@@ -0,0 +1,220 @@
+package tinydraw
+
+import (
+	"fmt"
+	"image"
+	"image/color"
+	"image/png"
+	"math"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+)
+
+type testDisplay struct {
+	*image.RGBA
+}
+
+func (d *testDisplay) Size() (x, y int16) {
+	return int16(d.Rect.Dx()), int16(d.Rect.Dy())
+}
+
+func (d *testDisplay) SetPixel(x, y int16, c color.RGBA) {
+	d.Set(int(x), int(y), c)
+}
+
+func newTestDisplay() testDisplay {
+	i := image.NewRGBA(image.Rect(0, 0, 320, 240))
+	return testDisplay{
+		RGBA: i,
+	}
+}
+
+func compareWithReference(name string, d testDisplay) error {
+
+	referenceFilePath := filepath.Join("testdata", name+".png")
+
+	referenceFile, err := os.Open(referenceFilePath)
+	switch err.(type) {
+	case *os.PathError:
+		saveImgFile, err := os.Create(referenceFilePath)
+		if err != nil {
+			return err
+		}
+
+		err = png.Encode(saveImgFile, d.RGBA)
+		return err
```

Comment:
If there is no expected value, then the test should fail.

Reply:
I added a flag to enable generation of new reference images to make creating new test easier. By default the tests will now fail if an image is missing